amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/repo/action/ExtendedActionServiceImpl.java

```diff
@@ -49,12 +53,17 @@
  */
 public class ExtendedActionServiceImpl extends ActionServiceImpl implements ApplicationContextAware
 {
+    private static final Set<String> ALLOWED_CONTEXTS_FOR_RM_ACTIONS = Set.of("rules");
+
     /** File plan service */
     private FilePlanService filePlanService;
 
     /** Application context */
     private ApplicationContext extendedApplicationContext;
 
+
+    private Set<String> recordManagementActions;
```

Comment:
Volatile of AtomicReference could be used to avoid possibility of other thread not seeing changes from other thread.